hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieBaseFileGroupRecordBuffer.java

```diff
@@ -495,14 +502,24 @@ protected boolean hasNextBaseRecord(T baseRecord, Pair<Option<T>, Map<String, Ob
     Map<String, Object> metadata = readerContext.generateMetadataForRecord(
         baseRecord, readerSchema);
 
-    Option<T> resultRecord = logRecordInfo != null
-        ? merge(Option.of(baseRecord), metadata, logRecordInfo.getLeft(), logRecordInfo.getRight())
-        : merge(Option.empty(), Collections.emptyMap(), Option.of(baseRecord), metadata);
-    if (resultRecord.isPresent()) {
-      nextRecord = readerContext.seal(resultRecord.get());
-      return true;
+    if (logRecordInfo != null) {
+      Option<T> resultRecord = merge(Option.of(baseRecord), metadata, logRecordInfo.getLeft(), logRecordInfo.getRight());
+      if (resultRecord.isPresent()) {
+        // Updates
```

Comment:
It does not look right to only do the stats while iterating over the base file records, the log file could also contain inserts/updates/deletes. If this logic is in sync with the old reader, we should be good now, just need to fix it in the future.

Reply:
lets file all these in followup JIRAs for 1.0.1. but good to fix them here if we can..

Reply:
@danny0405 this logic is in sync with the older compaction / merge handle logic, i.e., log files are merged first through the log record scanner, then the records from the base file and the merged records from the log scanner are routed to the merge handle for writing and stats.  The new file group reader and stats follow the exact same semantics to avoid behavior change.

Reply:
> the HoodieMergeHandle has a stat for insertRecordsWritten, we should be good here.
Yes, it is handled.